File: java/SR-RBD/src/ddb/restore/msg/RestoreTable.java
package ddb.restore.msg;

import java.io.IOException;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.msg.MessageType;

public class RestoreTable extends RestoreMessage{

	protected int tableVersion; 
	protected String tableName;
	protected String tableDump;
	
	public RestoreTable()
	{
		
	}
	
	public RestoreTable(int version, String name, String dump) {
		setTableVersion(version);
		setTableName(name);
		setTableDump(dump);
	}
	
	public void setTableName(String n) {
		tableName = n;
	}
	
	public int getTableVersion() {
		return tableVersion;
	}
	
	public void setTableVersion(int v) {
		tableVersion = v;
	}
	
	public String getTableName() {
		return tableName;
	}
	
	public void setTableDump(String d) {
		tableDump = d;
	}
	
	public String getTableDump() {
		return tableDump;
	}
	
	@Override
	protected void fromBinary(DataInputStream s) throws IOException {
		tableVersion = s.readInt();
		tableName = s.readString();
		tableDump = s.readString();
	}

	@Override
	public MessageType getType() {
		return MessageType.RESTORE_TABLE;
	}

	@Override
	protected void toBinary(DataOutputStream s) throws IOException {
		s.writeInt(tableVersion);
		s.writeString(tableName);
		s.writeString(tableDump);
	}

	

}

File: C#/RBD/RBD/Restore/Msg/RestoreTable.cs
// + TODO check

using RBD.Communication;

namespace RBD.Restore.Msg
{
    public class RestoreTable : RestoreMessage {

	    protected int tableVersion; 
	    protected string tableName;
	    protected string tableDump;
    	
	    public RestoreTable()
	    {
    		
	    }
    	
	    public RestoreTable(int version, string name, string dump) {
		    setTableVersion(version);
		    setTableName(name);
		    setTableDump(dump);
	    }
    	
	    public void setTableName(string n) {
		    tableName = n;
	    }
    	
	    public int getTableVersion() {
		    return tableVersion;
	    }
    	
	    public void setTableVersion(int v) {
		    tableVersion = v;
	    }
    	
	    public string getTableName() {
		    return tableName;
	    }
    	
	    public void setTableDump(string d) {
		    tableDump = d;
	    }
    	
	    public string getTableDump() {
		    return tableDump;
	    }

        override public void FromBinary(DataInputStream s)
        {
		    tableVersion = s.ReadInt32();
		    tableName = s.ReadString();
		    tableDump = s.ReadString();
	    }

        override public MessageType GetMessageType()
        {
		    return MessageType.RESTORE_TABLE;
	    }

        override public void ToBinary(DataOutputStream s)
        {
            s.Write((int)tableVersion);
		    s.WriteString(tableName);
		    s.WriteString(tableDump);
	    }
    }
}
